src/index.ts

```diff
@@ -6,18 +6,23 @@ import { normalizeOptions } from './normalizeOptions'
 import { hashObject } from './utils'
 
 let cacheOptionsHash: string | undefined
-let resolver: ResolverFactory | undefined
-export function resolve(source: string, file: string, options?: NapiResolveOptions | null): { found: boolean, path: string | null | undefined } {
+let cachedResolver: ResolverFactory | undefined
+
+export function resolve(source: string, file: string, options?: NapiResolveOptions | null, resolver: ResolverFactory | null = null): { found: boolean, path: string | null | undefined } {
   if (isBuiltin(source))
     return { found: true, path: null }
 
-  options ??= {}
-  const optionsHash = hashObject(options)
+  if (resolver == null) {
```

Comment:
With interface version 3, the `resolver` will not be null, so all those hashing and caching below would be skipped~